extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/PreInitRecorder.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.amazon.lambda.runtime;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+/**
+ * The class pre-initialization recorder.
+ */
+@Recorder
+public class PreInitRecorder {
+    public PreInitRecorder() {
+    }
+
+    public void preInit() {
+        final InputStream is = PreInitRecorder.class.getResourceAsStream("/META-INF/pre-init-classes.txt");
```

Comment:
A lot easier for extensions and libraries to just create this file manually than doing the overhead of a BuildItem.  I envision multiple pre-init-classes.txt files being shipped with different extensions and just loading them if needed.

Reply:
Also imagine having writing 2000 lines of code all to allocate BuildTimeInitializedClassBuildItem instances.  If you're thinking of generating this by just adding the application classes, that is not not nearly enough.  Lambda for instance, loads 243 classes at runtime with a simple hello world example.

Reply:
In general extension has a deeper knowledge on the class that would be used. For classes always loaded, fine, but there are cases where we only know them at build time once the extension has looked  into the app. 

Also for the application code, how would you list these classes and the arc proxies? 

We can have both mechanism, the txt files would contain the static set (most of the classes I believe), while the build item will be for the more dynamic set.

WDYT?

Reply:
I'll just do this work ok?  Just want to get it done so we can share and do lambda testing.